tests/handlers/test_appservice.py

```diff
@@ -597,16 +597,22 @@ def test_sending_read_receipt_batches_to_application_services(self) -> None:
 
         # Now, pretend that we receive a large burst of read receipts (300 total) that
         # all come in at once.
-        for i in range(300):
+        for _ in range(300):
+            room_id = self.helper.create_room_as(
+                self.local_user, tok=self.local_user_token
+            )
+            resp = self.helper.send(room_id, tok=self.local_user_token)
+            event_id = resp["event_id"]
+
```

Comment:
Creating a room in between each read receipt does a lot of stuff, including creating new events. I'm not sure if this test is still testing the intended scenario.

Reply:
Agreed, this is also very slow, not sure how to proceed here. The original test specifies a unique room/event per receipt so I was trying to replicate that...